support/ruby/consolr/lib/consolr.rb

```diff
@@ -0,0 +1,188 @@
+      rescue Exception => e
+        puts e
+        puts "-------"
+        puts "Dangerous Status -- #{@dangeous_status}"
+        puts "Please specify the statuses which are dangerorous, during which dangerous shouldn't be run."
+        puts "-------"
+        exit 0
+      end
+
+      @dangerous_actions = [:off, :reboot] # Must match the symbol in options{}
+
+      @options = {}
+      @opt_parser = OptionParser.new do |opt|
+        opt.banner = "Usage: consolr [OPTIONS]"
+        opt.separator  ""
+        opt.separator  "Options"
+        
+        opt.on("-a","--asset ASSET","asset tag or hostname") { |asset| options[:asset] = asset }
+        opt.on("-o","--on","turn on node") { options[:on] = true }
+        opt.on("-x","--off","turn off node") { options[:off] = true }
+        opt.on("-r","--reboot","restart node") { options[:reboot] = true }
+        opt.on("-i","--identify","turn on chassis UID") { options[:identify] = true }
+        opt.on("-c","--console","console into node via SOL") { options[:console] = true }
+        opt.on("-k","--kick","kick if someone is hogging the console") { options[:kick] = true }
+        opt.on("-d","--dangerous","list dangerous stuff") { options[:dangerous] = true }
+        opt.on("-f","--force","force run dangerous actions") { options[:force] = true }
+        opt.on("-s","--sdr","Sensor Data Repository (SDR)") { options[:sdr] = true }
+        opt.on("-l","--log LOG","System Event Log (SEL) [list|clear]") { |log| options[:log] = log }
+        
+        opt.on("-h","--help","help") { puts @opt_parser }
+      end
+    end
+    
+    def ipmitool_cmd action
+      system("#{@ipmitool_exec} -I lanplus -H #{@node.ipmi.address} -U #{@node.ipmi.username} -P #{@node.ipmi.password} #{action}")
+      return $?.exitstatus == 0 ? "SUCCESS" : "FAILED"
+    end
+
+    def start
+      begin
+        @opt_parser.parse! # extract from ARGV[]
+        raise OptionParser::MissingArgument, "-a" if options[:asset].nil?
+      rescue Exception => e
+        puts @opt_parser
+        exit 0
+      end
+      
+      abort("Cannot find #{@ipmitool_exec}") unless File.exist?(@ipmitool_exec)
+
+      dangerous_body = "Dangerous actions: #{dangerous_actions.join(', ')}\n"\
+        "Dangerous status: #{dangerous_status.join(', ')} (override with --force)\n"\
+        "Dangerous assets: #{dangerous_assets.join(', ')} (ignored no matter what, even with --force)"
+      
+      if options[:dangerous]
+        puts dangerous_body
+        exit 0
+      end
+      
+      begin
+        collins = Collins::Authenticator.setup_client
+      rescue Exception => e
+        puts e
+        puts "-------"
+        puts "There was a problem setting up a connection with Collins."
+        puts "-------"
+        exit 0
+      end
+     
+      # match assets like vm-67f5eh, zt-*, etc.
+      nodes = options[:asset].match(/[a-zA-Z0-9\-\_]+/) ? (collins.find :tag => options[:asset]) : (collins.find :hostname => options[:asset])
```

Comment:
This will match `web-abcdef123` as a tag, not hostname. Why cant this be made explicit instead? I.e. have a `-t TAG` and `-H HOSTNAME` flags, because this match wont work for everyone.